test/Microsoft.AspNetCore.Cors.Test/UriExtensionsTests.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Cors.Infrastructure
+{
+    public sealed class UriExtensionsTests
+    {
+        [Theory]
+        [MemberData(nameof(IsSubdomainOfTestData))]
+        public void TestIsSubdomainOf(Uri subdomain, Uri domain, bool expectedResult)
+        {
+            Assert.Equal(subdomain.IsSubdomainOf(domain), expectedResult);
+        }
+
+        public static IEnumerable<object[]> IsSubdomainOfTestData
+        {
+            get
+            {
+                return new[]
+                {
+                    new object[] {new Uri("http://sub.domain"), new Uri("http://domain"), true},
```

Comment:
Per my previous comment, I'm guessing that this:

```c#
new object[] {new Uri("http://subdomain"), new Uri("http://domain"), true}, 
```

Would also pass, even though it is wrong. (`subdomain` is clearly not a sub-domain of `domain`.)